src/category_theory/category.lean

```diff
@@ -84,6 +84,28 @@ section
 variables {C : Sort u} [𝒞 : category.{v} C] {X Y Z : C}
 include 𝒞
 
+lemma eq_of_comp_left_eq {f g : X ⟶ Y} (h : ∀ {Z : C} (h : Y ⟶ Z), f ≫ h = g ≫ h) : f = g :=
```

Comment:
Lean doesn't get confused, but I do get confused by a hypothesis `h` that takes a morphism `h` as argument. (I don't use de Bruijn numbering internally :grinning:)

Reply:
Fixed.